pmlc/dialect/pxa/transforms/passes.h

```diff
@@ -13,10 +15,23 @@ class OpPassBase;
 
 namespace pmlc::dialect::pxa {
 
+struct StencilPassOptions
```

Comment:
I'm not sure we even need a separate Options object. But even if we did, I wouldn't have it use the `Option<unsigned> numThreads`, this should only be confined to options that are directly on a pass so that the parse pipeline infrastructure can bind command line arguments to the pass.

Reply:
The registration is taking an extra template parameter that is the Options object. The parser fills these struct with the parsed values.
The MLIR doc describing this is at https://github.com/llvm/llvm-project/blob/master/mlir/docs/WritingAPass.md#instance-specific-pass-options